addons/website/controllers/main.py

```diff
@@ -334,9 +334,24 @@ def website_configurator(self, step=1, **kwargs):
             action_url += '&step=' + str(step)
         return request.redirect(action_url)
 
+    def _get_social_default_values(self, social=None):
+        """ Return the default value for the given social network. If the social
+        network is not specified, return a dict with all the default values.
+        """
+        default_values = {
+            'facebook': 'https://www.facebook.com/Odoo',
+            'github': 'https://github.com/odoo',
+            'linkedin': 'https://www.linkedin.com/company/odoo',
+            'youtube': 'https://www.youtube.com/user/OpenERPonline',
+            'instagram': 'https://www.instagram.com/odoo.official',
+            'twitter': 'https://twitter.com/Odoo',
+            'tiktok': 'https://www.tiktok.com/@odoo',
+        }
+        return default_values.get(social) if social else default_values
+
     @http.route(['/website/social/<string:social>'], type='http', auth="public", website=True, sitemap=False)
     def social(self, social, **kwargs):
-        url = getattr(request.website, 'social_%s' % social, False)
+        url = getattr(request.website, 'social_%s' % social, False) or self._get_social_default_values(social)
```

Comment:
Can't we avoid the security warning with something like:
```py
field_name = f'social_{social}'
url = field_name in request.website and request.website[field_name] or self._get_social_default_values(social)
```

Reply:
We should not try to avoid security warning just to avoid security warning, it seems more nicely written with `getattr` on top of being less diff, right?

Reply:
Well, it's not "hack a way to avoid being detected", IMO it's applying their suggested alternative.
But let's ask for an override to minimize the diff. @odoo/rd-security

Reply:
Note that we could also just decide to not protect the route and do `request.website[f'social_{social}']`, but since it was like that initially and just a stable quick seo imp, I wouldn't do it

Reply:
> it's just a bit more convenient.

It's not about convenience, it's about the controller not returning a 500 error page if requested with a non existing social network. Maybe overkill, but seems like a very short / easy win? It would be easy to just turn it into a simple field access and let it crashes if unknown field, which is probably fine, but now that it was coded like that for ages.. Is it really worth changing?

> Note that if you absolutely don't want to change this you can just add the fallback as a separate assignment.

Do you mean :point_down: ?
```diff
-       url = getattr(request.website, 'social_%s' % social, False) or self._get_social_default_urls().get(social)
+       url = getattr(request.website, 'social_%s' % social, False)
+       if not url:
+           url = self._get_social_default_urls().get(social)
        if not url:
            raise werkzeug.exceptions.NotFound()
        return request.redirect(url, local=False)
```

Just to bypass the required security CI?
It seems to make the code worst just for the sake of bypassing it, isn't it?

Since the code was already there before (I am just adding a fallback), and since it's safe[^1], it's worth keeping it as is? Less diff, easier to read, less verbose

I'll apply the diff you suggest if you don't want to override the CI (or if there is a real security issue), no problem tho :shrug: 

[^1]: Well, one could access a `social_x` field on website which is not related to our social network set of fields, but if it's returning the info (can read it), it means there is other ways to get it too, and seems very unlikely

Reply:
@odoo/rd-security Would you be able to validate the CI or bounce on the current discussion if you really want to?

As:
- XMO is on timoff
- there is actually no security issue here?
- the diff of this pr is just adding a fallback on top of the discussed code (so not really related)
- I don't want to bypass the security ci by writing (IMHO) code uglier than it could be

It would be nice if it can be approved